_posts/2021-07-05-newsletter.md

```diff
@@ -0,0 +1,199 @@
+
+## Tree-sitter
+
+Neovim 0.5 adds **experimental** support for tree-sitter, a library that parses a piece of code into a syntax tree in an incremental and error-resilient way; this means that reparsing that code after an edit is very fast, and parsing error due to, e.g., typos remain localized and do not break parsing further down. This tree can then be efficiently queried to obtain syntax information about the code. This allows for improved and/or faster
+
+- syntax highlighting,
+- code navigation,
+- refactoring,
+- text objects and motions,
+- searching and replacing,
+
+and more.
+
+In addition, tree-sitter allows "injecting" trees for other languages, which makes it easy to support files that contain code in multiple languages. To learn more about tree-sitter, watch [Tree-sitter - A new parsing system for programming tools - Max Brunsfield](https://www.youtube.com/watch?v=Jes3bD6P0To&feature=youtu.be&t=232).
+
+The goal is to replace the current vim regular expression-based syntax with tree-sitter, not only for better and faster syntax highlighting but also for new and improved ways of structured text editing. Currently, Neovim embeds or links to the tree-sitter library and provides a (Lua) API for generating and querying the syntax tree. User-facing features like the ones mentioned above are implemented in plugins like
+
+- [Nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) -- Highlighting, folds, incremental selection.
+- [Playground](https://github.com/nvim-treesitter/playground) -- Utility functions to easily show the parsed tree and run queries against it.
+- [Nvim-treesitter-textobjects](https://github.com/nvim-treesitter/nvim-treesitter-textobjects) -- Better text objects for vim-style motions and operators.
+- [Nvim-refactor](https://github.com/nvim-treesitter/nvim-treesitter-refactor) -- Highlight definitions,  navigation, smart rename.
+- [Architext](https://github.com/vigoux/architext.nvim) -- structural text editing (i.e., syntax-aware search and replace).
+
+More information on Neovim's tree-sitter features can be found in [Nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) and read [`:h treesitter`](https://neovim.io/doc/user/treesitter.html); you should also check out Thomas Vigouroux's [Vimconf.live presentation](https://youtu.be/EReNOGuMBmo?t=515). 
+
+These plugins already work well in most situations, although there are still a number of bugs and performance regressions that may occur regularly and that need to be addressed before tree-sitter support in Neovim can be declared stable. Note also that enabling tree-sitter based highlighting currently disables the internal regex-based syntax engine _completely_ (which may break other features that rely on it). Fixing these issues and improving the API (which is not stable yet) will be a major focus of the development cycle leading up to the 0.6 release.
```

Comment:
`These plugins already work well in most situations` this has not been the case for the people around me or even myself. I fear this is going to incitate people try it only to be disappointed. We agreed that it should be presented as an experimental feature so we could tone down the excitement and start directly with `there are still a number of bugs and performance regressions`. Not to undermine the formidable effort done on treesitter but it's not as smooth as lsp yet.

Reply:
I completely agree; this is overselling. I would still like to start the sentence with a "nice thing" -- maybe "Although these plugins already work well in many/some situations, ..."?

Reply:
"Although these plugins already work well enough in some situations (file types and/or features), there are..."?

Reply:
(again with the parentheses ;))

Reply:
`work reasonably well `is too vague,  I wouldn't mention anything positive. We dont want to encourage people who were not going to try it to try it. but maybe that's just me. Comments on social media are really positive about treesitter but I wouldn't endorse it since we strive for more :D ! but maybe that's just me.

Reply:
"Vague" was precisely the goal ;) And I _would_ like to mention something positive (this statement is in fact true for some languages, like Lua).

We can take a poll: thumbs up -- start with vague nice words; thumbs down -- hit them straight up with the warning stick.

Reply:
alternative suggestions for "work" and "well" are also welcome.